modules/ip/control/address.c

```diff
@@ -87,22 +86,24 @@ static struct api_out addr_add(const void *request, void **response) {
 	if (ifaddrs->count == IP4_HOPLIST_MAX_SIZE)
 		return api_out(ENOSPC, 0);
 
-	if (ip4_nexthop_lookup(iface->vrf_id, req->addr.addr.ip, &nh_idx, &nh) == 0)
+	if (ip4_nexthop_lookup(iface->vrf_id, req->addr.addr.ip) != NULL)
 		return api_out(EADDRINUSE, 0);
 
-	if ((ret = ip4_nexthop_add(iface->vrf_id, req->addr.addr.ip, &nh_idx, &nh)) < 0)
-		return api_out(-ret, 0);
+	if ((nh = ip4_nexthop_get_new()) == NULL)
```

Comment:
Is there a case where we would need to get a new nh from the mempool without initialisizing it just after ?

I feel that the `mempool_get` could be hidden in `ip4_nexthop_init` call.

Reply:
Good point. I'll change that.